src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -761,7 +761,11 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     // The loop stops once the children have unmounted and error lifecycles are
     // called. Then we return to the regular flow.
 
-    if (capturedErrors !== null && capturedErrors.size > 0) {
```

Comment:
We decided that cleaning up `capturedErrors` when an error boundary's parent is unmounted (see test added in https://github.com/facebook/react/pull/10403/commits/894656cda2d60dac3b9de147c5bdf730c3e55b9d) wasn't worth it, and so `capturedErrors !== null` is not a reliable indicator that there are unhandled errors.

If you're referring to moving the check to a `while` loop, that's how it used to work, but we recently refactored the work loops to do as few checks as possible on each iteration, and only do certain checks (like whether the priority level is correct) right after committing, which is the only time they change. Requires a bit more duplicated code but fewer runtime checks.